labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,364 @@
+import math
+import logging
+from enum import Enum
+from typing import Optional, List, Tuple
+from concurrent.futures import ThreadPoolExecutor
+
+import requests
+import numpy as np
+import tensorflow as tf
+from retry import retry
+from pyproj import Transformer
+from pydantic import BaseModel, validator, conlist
+from pydantic.class_validators import root_validator
+
+from ..geometry import Point
+from .base_data import BaseData
+from .raster import RasterData
+"""TODO: consider how to swap lat,lng to lng,lt when version = 2...
+should the bounds validator be inside the TiledImageData class then, 
+since we need to check on Version?
+"""
+
+VALID_LAT_RANGE = range(-90, 90)
+VALID_LNG_RANGE = range(-180, 180)
+TMS_TILE_SIZE = 256
+MAX_TILES = 300  #TODO: thinking of how to choose an appropriate max tiles number. 18 seems too small, but over 1000 likely seems too large
+TILE_DOWNLOAD_CONCURRENCY = 4
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class EPSG(Enum):
+    """ Provides the EPSG for tiled image assets that are currently supported.
+    
+    SIMPLEPIXEL is Simple that can be used to obtain the pixel space coordinates
+
+    >>> epsg = EPSG()
+    """
+    SIMPLEPIXEL = 1
+    EPSG4326 = 4326
+    EPSG3857 = 3857
+    EPSG3395 = 3395
+
+
+class TiledBounds(BaseModel):
+    """ Bounds for a tiled image asset related to the relevant epsg. 
+
+    Bounds should be Point objects. Currently, we support bounds in EPSG 4326.
+
+    If version of asset is 2, these should be [[lat,lng],[lat,lng]]
+    If version of asset is 1, these should be [[lng,lat]],[lng,lat]]
+
+    >>> bounds = TiledBounds(
+        epsg=EPSG.4326,
+        bounds=[Point(x=0, y=0),Point(x=100, y=100)]
+        )
+    """
+    epsg: EPSG
+    bounds: List[Point]
+
+    @validator('bounds')
+    def validate_bounds_not_equal(cls, bounds):
+        first_bound = bounds[0]
+        second_bound = bounds[1]
+
+        if first_bound == second_bound:
+            raise AssertionError(f"Bounds cannot be equal, contains {bounds}")
+        return bounds
+
+    #bounds are assumed to be in EPSG 4326 as that is what leaflet assumes
+    @root_validator
+    def validate_bounds_lat_lng(cls, values):
+        epsg = values.get('epsg')
+        bounds = values.get('bounds')
+        #TODO: look into messaging that we only support 4326 right now. raise exception, not implemented
```

Comment:
@jtsodapop this is included in TiledBounds class, should consider if we should include it elsewhere